api/jsonapi.go

```diff
@@ -3803,6 +3803,27 @@ func (i *jsonAPIHandler) GETIPNS(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprint(w, string(retBytes))
 }
 
+func (i *jsonAPIHandler) ResolveIPNS(w http.ResponseWriter, r *http.Request) {
+	_, peerID := path.Split(r.URL.Path)
+
+	pid, err := peer.IDB58Decode(peerID)
+	if err != nil {
+		ErrorResponse(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*180)
```

Comment:
The timeout for this call is 2 minutes. Is this intended? /cc @drwasho

Reply:
Yeah I need it to be at least 2 minutes but we could make it configurable if somebody only wanted entries that resolve quickly.

Reply:
This is no problem for me, I just want to make sure this was intended.